src/status_im/utils/notifications.cljs

```diff
@@ -16,9 +16,24 @@
  (fn [db [_ fcm-token]]
    (assoc-in db [:notifications :fcm-token] fcm-token)))
 
+(handlers/register-handler-fx
+ :request-notifications-granted
+ (fn [_ _]))
+
+(handlers/register-handler-fx
+ :request-notifications-denied
+ (fn [_ _]))
+
 ;; NOTE: Only need to explicitly request permissions on iOS.
 (defn request-permissions []
-  (.requestPermissions (.-default rn/react-native-fcm)))
+  (-> (.requestPermissions (.-default rn/react-native-fcm))
+      (.then
+        (fn [_]
+          (log/debug "notifications-granted")
+          (dispatch [:request-notifications-granted {}]))
```

Comment:
@jeluard what do you think, is it ok that we need to add events only for mixpanel? maybe we could dispatch something like `[:mixpanel :target-id]` ?

Reply:
I agree it's a little weird but that's probably the best option as it keeps the notification semantic. We may imagine other logic added later (like some popup) or other tracking (like instabug).